### PR TITLE
Fixed Issue #25128:Newsletter subscribe and unsubscribe issue for Customer

### LINKED ISSUE #25128: Newsletter subscribe and unsubscribe issue for Customer
### Preconditions (*)

1. Magento 2.3.3

### Steps to reproduce (*)

1. Login as customer on frontend.
2. Click on Newsletter Subscriptions link.
3. Login into admin panel.
4. Disable Newsletter. Navigate to Store -> Configuration -> Customer -> Newsletter -> General Options and set enable to **NO**
5. Now go to opened Newsletter Subscriptions page and try to subscribe or unsubscrbe.

### Expected result (*)

1. There should be error or no route page.

### Actual result (*)

1. Customer is able to subscribe or unsubscribe.

### PR DESCRIPTION
**Pull request by ghost**
### Fixed Issues (if relevant)

1. fixes magento/magento2#25128:Newsletter subscribe and unsubscribe issue for Customer

### Manual testing scenarios (*)

1. Login as customer on frontend.
2. Click on Newsletter Subscriptions link.
3. Login into admin panel.
4. Disable Newsletter. Navigate to Store -> Configuration -> Customer -> Newsletter -> General Options and set enable to **NO**
5. Now go to opened Newsletter Subscriptions page and try to subscribe or unsubscrbe.
6. Now there will be no route page 

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ghost** [comment]
@buskamuza @melnikovi - Please review the changes

File: app/code/Magento/Newsletter/Controller/Manage/Save.php [hunk 1]
```diff
@@ -88,7 +94,7 @@ public function execute()
                     ->getParam('is_subscribed', false);
                 if ($isSubscribedParam !== $isSubscribedState) {
                     // No need to validate customer and customer address while saving subscription preferences
-                    $this->setIgnoreValidationFlag($customer);
+                    $customer->setData('ignore_validation_flag', true);
```
Comment (melnikovi) [inline app/code/Magento/Newsletter/Controller/Manage/Save.php:105]: What is the reason for this change?

File: app/code/Magento/Newsletter/Controller/Manage.php [hunk 2]
```diff
@@ -6,29 +6,53 @@
...
      * Customer session
      *
      * @var \Magento\Customer\Model\Session
      */
     protected $_customerSession;
 
+    /**
+     * @var ScopeConfigInterface
+     */
+    protected $_scopeConfig;
```
Comment (melnikovi) [inline app/code/Magento/Newsletter/Controller/Manage.php]: Can we make these properties private and remove underscore from names?

**Comment by melnikovi** [comment]
Before merging this pull request I would recommend to check if any extensions on market palace extent Magento\Newsletter\Controller\Manage abstract controller.

File: app/code/Magento/Newsletter/Controller/Manage/Save.php (hunk 1 above)
Comment (ghost) [reply app/code/Magento/Newsletter/Controller/Manage/Save.php:105]: @melnikovi - It was throwing PHPMD coupling between objects error. so i checked and found that we don't need this method because it is private method and used only once in scope. Also there was no calculation in method.
If you wish i can revert it and remove PHPMD error by suppressing it

File: app/code/Magento/Newsletter/Controller/Manage.php (hunk 2 above)
Comment (ghost) [reply app/code/Magento/Newsletter/Controller/Manage.php]: @melnikovi - Yes we can, Do you want me to update both variables in both classes? or you just want me to update $_scopeConfig ?

**Comment by ghost** [comment]
@melnikovi - It is confirmed that class **Magento\Newsletter\Controller\Manage** is not used in any other extension.

File: app/code/Magento/Newsletter/Controller/Manage.php (hunk 2 above)
Comment (melnikovi) [reply app/code/Magento/Newsletter/Controller/Manage.php]: All parameters in both classes.

**Comment by melnikovi** [comment]
Do you mean Magento modules or third party extensions? It would be good to combine this PR and https://github.com/magento/magento2/pull/25127. Please also see https://github.com/magento/magento2/pull/25127#issuecomment-544689889.

File: app/code/Magento/Newsletter/Controller/Manage.php (hunk 2 above)
Comment (ghost) [reply app/code/Magento/Newsletter/Controller/Manage.php]: @melnikovi - I have updated the scope of variables. 
But i think we should use protected variables in parent classes so that these variables can be used in child classes. Else we need to re-declare the variable in child class as in this PR i had to re-declare the $customerSession in child class.
What do you think?

**Comment by ghost** [comment]
@melnikovi  - I have checked this and this class is not used in any other extension.
Please review pull request and let me know there is any update required from my end.

File: app/code/Magento/Newsletter/Controller/Manage.php (hunk 2 above)
Comment (melnikovi) [reply app/code/Magento/Newsletter/Controller/Manage.php]: Sorry, I meant fix only parameter names we introduced.

Comment (ghost) [reply app/code/Magento/Newsletter/Controller/Manage.php]: @melnikovi - What should i do now? Do i need to restore the scope of variables to original scope.?
And do i need to restore the variable names to original names?

**Comment by ghost** [comment]
@melnikovi - Is there any update over this PR?

**Comment by sdzhepa** [comment]
Closed after reviewing on internal Triage meeting
See for more details: https://github.com/magento/magento2/issues/25128#issuecomment-635563571
